### PR TITLE
fix(TDKN-177): fix nested properties post deserialization

### PR DESCRIPTION
**Pull request by akhabali**
**What is the problem this Pull Request is trying to solve?**
Post deserialization of nested properties is not working
 
**What is the chosen solution to this problem?**
Collect the post deserialization handler correctly 
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-177
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/213/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/213/)

File: daikon/src/test/java/org/talend/daikon/serialize/migration/PropertyMigrationTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
...org.talend.daikon.serialize.migration.ParentProperties\",\"versionZeroProp\":{\"@type\":\"org.talend.daikon.properties.property.StringProperty\",\"possibleValues2\":null,\"flags\":null,\"storedValue\":\"I'm born since version 0\",\"children\":{\"@type\":\"java.util.ArrayList\"},\"taggedValues\":{\"@type\":\"java.util.HashMap\"},\"size\":-1,\"occurMinTimes\":0,\"occurMaxTimes\":0,\"precision\":0,\"pattern\":null,\"nullable\":false,\"possibleValues\":null,\"currentType\":\"java.lang.String\",\"name\":\"versionZeroProp\",\"tags\":null},\"versionZeroNestedProp\":{\"__version\":1,\"versionOneProp\":{\"@type\":\"org.talend.daikon.properties.property.StringProperty\",\"possibleValues2\":null,\"flags\":null,\"storedValue\":\"I'm born since version 1\",\"children\":{\"@type\":\"java.util.ArrayList\"},\"taggedValues\":{\"@type\":\"java.util.HashMap\"},\"size\":-1,\"occurMinTimes\":0,\"occurMaxTimes\":0,\"precision\":0,\"pattern\":null,\"nullable\":false,\"possibleValues\":null,\"currentType\":\"java.lang.String\",\"name\":\"versionOneProp\",\"tags\":null},\"name\":\"versionZeroNestedProp\",\"validationResult\":null,\"tags\":null},\"name\":\"parent\",\"validationResult\":null,\"tags\":null}";
```
Comment (sgandon) [inline daikon/src/test/java/org/talend/daikon/serialize/migration/PropertyMigrationTest.java]: I suppose you should remove versionOne from the serialized string to only have version zero.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/214/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/214/)

File: daikon/src/test/java/org/talend/daikon/serialize/migration/PropertyMigrationTest.java (hunk 1 above)
Comment (akhabali) [inline daikon/src/test/java/org/talend/daikon/serialize/migration/PropertyMigrationTest.java]: This way it test the migration of nested & non nested properties __with different versions__.
The use case is :
* Parent
   * property1(simple)   => version 0
    * property2(Nested)  => version 0 of parent and version 1 of it self 
    * property3(simple)   => version 1 _(need migration)_
* Property2(Nested)
    * property1(simple)  => version 1
    * property2(simple)  => version 2 _(need migration)_

This way it grantee the good migration of different versions of properties. seems working well.

Comment (sgandon) [inline daikon/src/test/java/org/talend/daikon/serialize/migration/PropertyMigrationTest.java]: ok I got it, thanks.

**Comment by akhabali** [comment]
We have a postDeserialize life cycle issue here !
What happens when we have nested properties.
-> parent.postDeserialize() get called
-->   parent.initLayout()
----> child . initLayout()
---------> child . setupLayout();   __postDeserialize() of child have not been called yet. NPE issue__
---------> child . refreshAllFormsLayout();  __postDeserialize()  of child have not been called yet. NPE issue__

https://github.com/Talend/daikon/blob/82745bb87c4aa5c15dc9c3945d828a7165513144/daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java#L74

File: daikon/src/main/java/org/talend/daikon/properties/Properties.java [hunk 2]
```diff
@@ -113,6 +114,11 @@
     Properties init();
 
     /**
+     * setup layout and refresh all it forms
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/properties/Properties.java]: As mentionned in Slack I prefer you do not change this interface but rather use the init() method after updating the `postDeserialized()` with a `propsAlreadyInitialized=true`.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/215/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/215/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/216/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/216/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/224/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/224/)

**Pull request merged**